src/AppBundle/Controller/BlogController.php

```diff
@@ -114,4 +116,43 @@ public function commentFormAction(Post $post)
             'form' => $form->createView(),
         ));
     }
+
+    /**
+     * @Route("/search", name="blog_search")
+     * @Method("GET")
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $q = $request->query->get('q');
```

Comment:
Well, variable's name is matching with input name, so I prefer to use `$q` to be consistent.